### DISCUSSION TITLE
Angular E2E Production Builds

### DISCUSSION THREAD

**Discussion by Nico-VanHaaster** [cypress-io/cypress#25927] [Questions and Help]
Hello,  I am very new to Cypress so please be gentle.

The issue I am having is that when I run my E2E tests using the Angular Dev Server then the tests seem to work fine.
However when pointing to our production environment the visit(..) loads the page but does not seem to work with our angular routing. 

For example here would be the process
1. visit('/pageprotected')
2. browser redirected to '?returnUrl=/pageProtected'
3. Additional work carried out and then redirected back to '/pageprotected'

This works fine in the dev environment however in a production environment (build). After step 2 nothing happens, I have seen that our Angular Guard does fire and cause the redirect, however the component that completes step 3 does not get constructed.

Our configfuration is nothing more than the out of the box configuration with a single cy.visit() statement.

**Comment by Nico-VanHaaster** [comment]
I found the config flag

experimentalModifyObstructiveThirdPartyCode: true

Adding this seems to have changed things. Not sure what it does but will see if this fixes things. I guess one thing we do have is azure based authentication which may launch these frames.

**Reply by MaryNuella** [reply]
Please do you happen to know how I can select an option from a drop-down in Angular???

 Please help
